nodeadm/internal/kubelet/config.go

```diff
@@ -240,6 +240,38 @@ func (ksc *kubeletConfig) withDefaultReservedResources() {
 	ksc.KubeReservedCgroup = ptr.String("/runtime")
 }
 
+// withPodInfraContainerImage determines whether to add the
+// '--pod-infra-container-image' flag, which is used to ensure the sandbox image
+// is not garbage collected.
+//
+// TODO: revisit once the minimum supportted version catches up or the container
+// runtime is moved to containerd 2.0
+func (ksc *kubeletConfig) withPodInfraContainerImage(cfg *api.NodeConfig, kubeletVersion string, flags map[string]string) error {
+	// the flag is a noop on 1.29+, since the behavior was changed to use the
+	// CRI image pinning behavior and no longer considers the flag value.
```

Comment:
IIUC, we also need to add https://github.com/awslabs/amazon-eks-ami/pull/1605 into al2023 to help with 1.29 right?

Reply:
yea, it looks like we will need the `crictl` for sure in some equivalent fashion. Am i missing any go libraries to pull the image through the CRI versus `crictl`?

Reply:
i could use the same upstream resources to pull this to remove the cri-tools install,
it would look something like:
```
imageManager, err := remote.NewRemoteImageService("unix:///run/containerd/containerd.sock", 2*time.Second, nil)
imageSpec := v1.ImageSpec{Image: sandboxImage}
authConfig := v1.AuthConfig{Auth: ecrUserToken}
imageRef, err := imageManager.PullImage(context.TODO(), &imageSpec, &authConfig, nil)
```

Reply:
Chatted offline, I think it's a good idea for `nodeadm` to bake in this functionality but we need to find the right CRI client library to use (pulling in kubelet internals should be a last resort)